### PR TITLE
Refactor the TLS configuration between meta.Client and the api.Config

### PR DESCRIPTION
**Pull request by dadgar**
This PR refactors the TLS configuration between meta.Client and the api.Config such that users of the API package can also reuse the TLS configuration logic.

### CODE REVIEW

File: api/client.go [hunk 1]
```diff
@@ -93,16 +138,12 @@ func (c *Config) ReadEnvironment() error {
 	var envCAPath string
 	var envClientCert string
 	var envClientKey string
-	var envInsecure bool
-	var foundInsecure bool
+	envInsecure := true
```
Comment (vishalnayak) [inline api/client.go]: Shouldn't this be `false` by default?

File: api/client.go [hunk 2]
```diff
@@ -84,6 +84,51 @@ func DefaultConfig() *Config {
 	return config
 }
 
+// ConfigureTLS takes the CA, client and TLS configuration and configures the
+// the HTTP client TLS config.
+func (c *Config) ConfigureTLS(CACert, CAPath, clientCert, clientKey,
```
Comment (jefferai) [inline api/client.go]: These values should be put into a struct, rather than an endless number of parameters.

File: api/client.go (hunk 1 above)
Comment (jefferai) [inline api/client.go]: Yes, should definitely be false. Also, as a coding style thing, please just use `var envInsecure bool` to match the rest.

**Comment by jefferai** [comment]
:ship:

**Pull request merged**